OpenSim/Framework/Servers/BaseOpenSimServer.cs

```diff
@@ -311,12 +311,21 @@ public virtual void Startup()
         /// </summary>      
         public virtual void Shutdown()
         {
+            Shutdown(0);
```

Comment:
IMO a clean exit (shutdown OR restart) should return 0.  Anything else is a non-zero return.   Where anything else is generally reserved for a crash or unusual termination (process killed).  Returning "state" in an exit code now requires a caller to check it.  And IMO that's bad practice.